types/koa/index.d.ts

```diff
@@ -711,9 +715,10 @@ declare namespace Application {
         respond?: boolean;
     }
 
-    type ParameterizedContext<StateT = DefaultState, CustomT = DefaultContext> = ExtendableContext & {
-        state: StateT;
-    } & CustomT;
+    type ParameterizedContext<StateT = DefaultState, ContextT = DefaultContext, ResponseBodyT = unknown> = ExtendableContext
+        & { state: StateT; }
+        & ContextT
+        & { body: ResponseBodyT; response: { body: ResponseBodyT }; };
```

Comment:
overriding these properties opens up the potential for being inconsistent, since these don't necessarily need to correlate with the existing reference to `body` and `response.body`.

Reply:
There is a chance a middleware could modify the body, that's true, yet one could provide a type for what body could be populated before with... It's optional for the developer to define strict type here...

Reply:
I still don't find this to be appropriate to exist within this types package, as far as Koa is concerned, the response body can and should still be any.

This type enforcement can be handled within your application by extending the Koa types already

Reply:
That's correct. And so far with this PR it's still `any` by default, but allows developer to provide more specific type if he wants. Check line [537](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50249/files#diff-bb94074d9ae1164d531dd6eb17a8858f06f4129662469a7d5e1dd19cb5eed86cR537)

```
type Middleware<StateT = DefaultState, ContextT = DefaultContext, ResponseBodyT = any> = compose.Middleware<
```

With this set-up I would be able to define my middleware as:

```
interface MyMiddlewareResponse {
  prop: string
}

export const myMiddleware: Middleware<{}, {}, MyMiddlewareResponse> = (ctx, next) => { ... }
```

Similarly like you can provide a more specific type for state and context. Is there anything wrong with it? I don't think so. Please tell me exactly what it would be? If you don't specify the third param for the `Middleware` it will be `any` by default, so this PR does not change current behaviour at all.

If there is another way to achieve the example I have shown above, meaning easily provide a type for response body, please could you provide an example?

Thank you in advance for your time :)